PathFinder.cc

```diff
@@ -11,38 +11,47 @@ namespace PathFinder {
 
 std::vector<PxRigidActor *> actors;
 
-float voxelHeight = 1.5;
-float voxelHeightHalf = voxelHeight / 2;
-float heightTolerance = 0.6;
-Vec start;
-Vec dest;
-float detectStep = 0.1;
-unsigned int iterDetect = 0;
-unsigned int maxIterDetect = 2000;
+Vec localVectorGetPath;

```

Comment:
I don't think these should be globals.

If `PathFinder` is a C++ object, like Scene, then these things can be instance variables.

For C++ there is really not much point to these shared global variables since C++ is already good at optimizing locals on the stack. You can just declare them when used.

Reply:
> For C++ there is really not much point to these shared global variables, since C++ is already good at optimizing locals on the stack. You can just declare them when used.

https://github.com/webaverse/app-wasm/pull/35/commits/7ab6e2eedc1add4bd9b2f2b6432545910f72d790 , not sure whether my understand is correct.

Reply:
> If PathFinder is a C++ object, like Scene, then these things can be instance variables. You can pass back PathFinder address to JS and pass it to WASM (+ cast to PathFinder *) to use it.

I'm not very clear about this. Should I only create `PathFinder` instance once and pass it back and forth between js and wasm?
But, this will cause need to `reset()` every time again?